packages/runtime/src/interpreter/opcode.ts

```diff
@@ -133,6 +135,20 @@ export class Op {
     return b;
   }
 
+  /**
+   * asserts if given variable type is address
+   * @param a - value to assert (should in bytes)
+   * @param line - line number in TEAL file
+   */
+  assertAddress (a: unknown, line: number): Uint8Array {
+    const bytes = this.assertBytes(a, line);
+    const addr = encodeAddress(bytes);
+    if (!isValidAddress(addr)) {
+      throw new RuntimeError(RUNTIME_ERRORS.TEAL.INVALID_ADDR, { addr: a, line: line });
+    }
```

Comment:
i don't follow this logic -- i think `encodeAddress` will fail if `bytes` is not a valid algorand address. You try to encode first, and then validate.

Reply:
> i don't follow this logic -- i think `encodeAddress` will fail if `bytes` is not a valid algorand address. You try to encode first, and then validate.

If `bytes` is present for an invalid address `encodeAddress` still return answer but when we check with `isValidAddress` It's will return false.

Reply:
> If bytes is present for an invalid address encodeAddress still return answer

is this tested? So something like `encodeAddress(strToBytes("Hello World"))` doesn't break & return an answer/address?

Reply:
> > If bytes is present for an invalid address encodeAddress still return answer
> 
> is this tested? So something like `encodeAddress(strToBytes("Hello World"))` doesn't break & return an answer/address?
isValidAddress address will return false

Reply:
> > If bytes is present for an invalid address encodeAddress still return answer
> 
> is this tested? So something like `encodeAddress(strToBytes("Hello World"))` doesn't break & return an answer/address?

yes and `isValidAddress` address will return false